File: backend/src/main/java/dev/fos/model/Recall.java
package dev.fos.model;

/**
 * Auto-avaliação ao registrar um drill, convertida em nota de qualidade do SM-2.
 *
 * <p>O SRS aqui não agenda revisão de teoria, e sim de <em>drill</em>: o app diz o que treinar hoje
 * (docs/04-arvore-curriculo-bjj.md). A escala é curta de propósito — quanto mais opções, menos
 * consistente a auto-avaliação.
 */
public enum Recall {
    /** Não lembrava como fazer. */
    FORGOT(2),
    /** Lembrava, mas travou / saiu errado. */
    HARD(3),
    /** Saiu, com esforço. */
    OK(4),
    /** Saiu limpo, sem pensar. */
    EASY(5);

    private final int quality;

    Recall(int quality) {
        this.quality = quality;
    }

    /** Nota 0–5 do SM-2. */
    public int quality() {
        return quality;
    }
}

File: web/src/content/recall.ts
import type { Recall } from '@fos/domain';

/**
 * Como a auto-avaliação do drill é nomeada na tela.
 *
 * Vive aqui, e não dentro do `DrillForm`, porque dois lugares a usam: o formulário, para escolher,
 * e o histórico do nó, para reler. Rótulo duplicado é rótulo que diverge — e aqui a divergência
 * seria a mesma nota aparecendo com dois nomes na mesma página.
 */
export const RECALL_LABELS: ReadonlyArray<{ value: Recall; label: string; hint: string }> = [
  { value: 'FORGOT', label: 'Não lembrava', hint: 'Precisou reaprender do zero' },
  { value: 'HARD', label: 'Travei', hint: 'Lembrava, mas saiu errado' },
  { value: 'OK', label: 'Saiu', hint: 'Funcionou, com esforço' },
  { value: 'EASY', label: 'Saiu limpo', hint: 'Sem pensar' },
];

/**
 * O histórico vem do backend com o nome do enum. Valor desconhecido cai de volta nele próprio em
 * vez de sumir: um recall novo no backend não pode apagar a linha inteira da tela.
 */
export function recallLabel(recall: string | undefined): string {
  if (!recall) return '';
  return RECALL_LABELS.find((option) => option.value === recall)?.label ?? recall;
}
